google-cloud-errors/lib/google/cloud/errors.rb

```diff
@@ -118,6 +118,59 @@ def status_details
         cause.status_details
       end
 
+      ##
+      # Returns the `::Google::Rpc::ErrorInfo` object present in `status_details` array,
+      # given that the following is true:
+      #   * `status_details` exists and is an array
+      #   * there is exactly one `::Google::Rpc::ErrorInfo` object in the `status_details` array
+      #
+      # @return [::Google::Rpc::ErrorInfo, nil]
+      def error_info
+        return nil unless status_details.is_a? Array
+        error_infos = status_details.find_all { |status| status.is_a?(::Google::Rpc::ErrorInfo) }
+        return nil unless error_infos.length == 1
+        error_infos[0]
```

Comment:
Consider memoizing this so the logic doesn't get rerun every time `error_info` is called. (Note that the other methods below already call `error_info` twice each.)

Reply:
:+1: